source/adapters/level_zero/event.cpp

```diff
@@ -1205,6 +1225,25 @@ ur_result_t _ur_ze_event_list_t::createAndRetainUrZeEventList(
       this->UrEventList = new ur_event_handle_t[EventListLength];
     }
 
+    auto WaitListEmptyOrAllEventsFromSameQueue = [CurQueue, EventListLength,
```

Comment:
This is copy pasted from this exact same file. Might be worth making this a function.